ui/components/src/TimeChart/TimeChart.tsx

```diff
@@ -208,9 +205,11 @@ export const TimeChart = forwardRef<ChartInstance, TimeChartProps>(function Time
       dataset.push({ id: index, source: [...values], dimensions: ['time', 'value'] });
     });
 
+    const updatedSeriesMapping = enablePinning ? [...seriesMapping, pinnedCrosshair] : seriesMapping;
```

Comment:
Should `updatedSeriesMapping` be used instead of `seriesMapping` throughout this file now? If so, I think you missed a few spots. I'll note the ones I see.

Reply:
do we also need to check if `pinnedCrosshair` is not null here?

Reply:
Yeah I’ll confirm, I started a branch to adjust [reset behavior](https://github.com/perses/perses/pull/1317#issuecomment-1650545480) now that approach has changed, converting this PR back to draft